### DISCUSSION TITLE
In the optimize_acqf function, if the underlying minimizer fails in a constrained optimization problem, is it possible for candidates that violate the inequality constraints to be returned?

### DISCUSSION THREAD

**Discussion by alan-tan-wm** [meta-pytorch/botorch#908] [Q&A]
Hi, I have a question. If for some reason the underlying scipy minimizer (SLSQP) fails in a constrained optimization problem, is it possible for candidates that violate the constraints to leak through, or is there some sort of gate that has been implemented in botorch that will reject those candidates? 

Thanks!

**Comment by Balandat** [comment]
I guess it's possible - we only check the box bounds after generation here: https://github.com/pytorch/botorch/blob/main/botorch/generation/gen.py#L181-L183, but not not the linear constraints. Is constraint violation by SLSQP something that you've observed in the past?

**Reply by alan-tan-wm** [reply]
Thank you for the reply. 
I have observed inequality constraints violations when using BoTorch, but typically they have been minor (e.g. on the order of 5e-8). In working with SLSQP in the past (not with BoTorch), we have observed occasions where the optimizer fails (i.e. the return OptimizeResult object of the minimize function indicates that the optimization was not terminated successfully). 

Since the OptimizeResult object is not exposed outside of the optimize_acqf function, it is difficult to determine if the optimization completed successfully.
